include/sound/sof/header2.h

```diff
@@ -0,0 +1,410 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2021 Intel Corporation. All rights reserved.
+ */
+
+#ifndef __INCLUDE_SOUND_SOF_HEADER2_H__
+#define __INCLUDE_SOUND_SOF_HEADER2_H__
+
+#include <linux/types.h>
+#include <uapi/sound/sof/abi.h>
+
+/** \addtogroup sof_uapi uAPI
+ *  SOF uAPI specification.
+ *  @{
+ */
+
+/*
+ * IPC4 messages have two 32 bit identifier made up as follows :-
+ *
+ * header - msg type, msg id, msg direction ...
+ * extension - extra params such as msg data size in mailbox
+ *
+ * These are sent at the start of the IPC message in the mailbox. Messages
+ * should not be sent in the doorbell (special exceptions for firmware).
+ */
+
+enum MsgTarget {
+	/* Global FW message */
+	FW_GEN_MSG = 0,
+
+	/* Module message */
+	MODULE_MSG = 1
+};
+
+enum global_msg {
+	GLB_BOOT_CONFIG = 0,
+	GLB_ROM_CONTROL = 1,
+	GLB_IPCGATEWAY_CMD = 2,
+
+	/* Create EDF task and run RTOS instance in it */
+	GLB_START_RTOS_EDF_TASK = 3,
+	/* Stop RTOS and delete its EDF task context */
+	GLB_STOP_RTOS_EDF_TASK = 4,
+
+	GLB_PERF_MEASUREMENTS_CMD = 13,
+	GLB_CHAIN_DMA = 14,
+
+	GLB_LOAD_MULTIPLE_MODULES = 15,
+	GLB_UNLOAD_MULTIPLE_MODULES = 16,
+
+	/* pipeline settings */
+	GLB_CREATE_PIPELINE = 17,
+	GLB_DELETE_PIPELINE = 18,
+	GLB_SET_PIPELINE_STATE = 19,
+	GLB_GET_PIPELINE_STATE = 20,
+	GLB_GET_PIPELINE_CONTEXT_SIZE = 21,
+	GLB_SAVE_PIPELINE = 22,
+	GLB_RESTORE_PIPELINE = 23,
+
+	/* Loads library (using Code Load or HD/A Host Output DMA) */
+	GLB_LOAD_LIBRARY = 24,
+	GLB_INTERNAL_MESSAGE = 26,
+
+	/* Notification (FW to SW driver) */
+	GLB_NOTIFICATION = 27,
+	GLB_MAX_IXC_MESSAGE_TYPE = 31
+};
+
+/* Message direction */
+enum MsgDir {
+	MSG_REQUEST = 0,
+	MSG_REPLY = 1,
+};
+
+enum PipelineState {
+	PIPE_RESET = 2,
+	PIPE_PAUSED = 3,
+	PIPE_RUNNING = 4,
+	PIPE_EOS = 5
+};
```

Comment:
you will need prefixes for all the enum definitions, e.g. sof_ipc_pipeline_state, SOF_IPC_PIPE_RESET

What you have for now is way too generic and conflict prone.

Reply:
ack